### DISCUSSION TITLE
Ollama + Llama 3 with planners and native functions

### DISCUSSION THREAD

**Discussion by dluc** [microsoft/semantic-kernel#6624] [Q&A] [answered]
From Javier on Discord:

> Hi Devs, I tried some experiments using SK with local LLM (Ollama + Llama 3). The chat completion works fine, but we can't use planners and native functions. Does anyone know if it's possible to use a local LLM for this purpose like LangChain does?

src: https://discord.com/channels/1063152441819942922/1075907964667437072/1248661245515534406

**Comment by sophialagerkranspandey** [comment, marked as answer]
Llama 3 doesn't support function calling from the model. If this is .NET you can use the handlebars planner - however in the long run, the handlebars planner will be retired.

**Comment by amrlafi** [comment]
@sophialagerkranspandey What will replace handlebars planner?
